andela/banshee-bn-backend
# #167891578 Implement Reset Password Feature

#### What does this PR do?
Implements reset password via email feature
#### Description of Task to be completed?
* Write tests
* Write email and password validations
* Create forgot password and reset password controllers
* Create helper function to send email with reset token to user
* Create forgot password and reset password routes
#### How should this be manually tested?
* Pull this branch ```ft-reset-password-167891578```
* Run ```npm run devstart``` command in the terminal
* Seed user into the database
* On postman, make a POST request to ```localhost:3000/api/v1/auth/forgot_password```
* Check the inbox of the user email you seeded into the database
* Get the reset token from the reset link sent and make a POST request with valid credentials to ```localhost:3000/api/v1/auth/reset_password/:{resetToken}```
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#167891578](https://www.pivotaltracker.com/story/show/167891578)
#### Screenshots (if appropriate)
![Screenshot (123)](https://user-images.githubusercontent.com/25702378/63839821-e6ae8900-c977-11e9-9e6d-6095ee3efcc5.png)

![Screenshot (122)](https://user-images.githubusercontent.com/25702378/63778773-5f143c00-c8dd-11e9-87b7-0b4a021ea950.png)
